chef/cookbooks/ntp/templates/default/ntp.conf.erb

```diff
@@ -31,9 +34,25 @@ server <%= ntp_server %> iburst minpoll 4
 # that might be intended to block requests from certain clients could also end
 # up blocking replies from your own upstream servers.
 
+<% if @is_server -%>
+# By default, ignore all inbound traffic from all hosts
+restrict -4 default ignore
+restrict -6 default ignore
+
+# Exchange time with all hosts in the admin subnet, but don't allow configuration.
+restrict <%= @admin_subnet %> mask <%= @admin_netmask %> notrap nomodify nopeer noquery
+
+# Exchange time with external NTP servers, but don't allow configuration
+# and use rate-limiting.
+<% @ntp_servers.each do |ntp_server| -%>
+restrict <%= ntp_server %> kod limited notrap nomodify nopeer noquery
+<% end -%>
+
+<% else -%>
 # By default, exchange time with everybody, but don't allow configuration.
-restrict -4 default kod notrap nomodify nopeer noquery
-restrict -6 default kod notrap nomodify nopeer noquery
+restrict -4 default notrap nomodify nopeer noquery
+restrict -6 default notrap nomodify nopeer noquery
```

Comment:
Delete the above 3 lines, as they override the default settings earlier in this change.

Reply:
@JanZerebecki , as I mentioned earlier, `kod` has no effect unless used in combination with `limited`, which is expressed as a warning in the logs:
```
KOD does nothing without LIMITED.
```
Also see [Reinhard's comment](https://bugzilla.suse.com/show_bug.cgi?id=1047941#c18) in the bug log.

Reply:
My suggestion has nothing to do with kod, but that these overwrite the ignore flag for default. From man ntp.conf: "last  match  found  defining  the  restriction  flags  associated with the entry."

Reply:
In that case, I think you're reading this wrong: there's an if/else block here that doesn't let that happen.